src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -438,7 +438,7 @@ module TcRecdUnionAndEnumDeclarations =
         rfspec
 
     let TcAnonFieldDecl cenv env parent tpenv nm (SynField(Attributes attribs, isStatic, idOpt, ty, isMutable, xmldoc, vis, m)) =
-        let id = (match idOpt with None -> mkSynId m nm | Some id -> id)
+        let id = match idOpt with None -> mkSynId ty.Range.StartRange nm | Some id -> id
```

Comment:
In theory we could also add `.MarkSynthetic()` on the range, and in general could use that to indicate compiler-generated identifiers.   However what you have is OK

Reply:
Yes, I was thinking about that too, but it should be a systematic approach to consider it here.

Reply:
Hm, there's an interesting test failure. I've fixed it by updating the baseline, but now I think it's not good that the produced error covers the empty range. I should probably indeed replace the approach with using a synthetic range here.